File: frontend/apps/web-tdesign/src/api/sales/market.ts
import { requestClient } from '#/api/request';

export interface MarketOverviewParams {
  year: number;
  month: number;
  level_type?: string;
  data_type?: 'production' | 'retail';
  date_type?: 'monthly' | 'quarterly' | 'yearly';
}

export interface MarketTrendParams {
  level_type?: string;
  years?: number;
  granularity?: 'monthly' | 'yearly';
  data_type?: 'production' | 'retail';
  date_type?: 'monthly' | 'quarterly' | 'yearly';
}

export interface MarketYearlyParams {
  year: number;
  level_type?: string;
  data_type?: 'production' | 'retail';
  date_type?: 'monthly' | 'quarterly' | 'yearly';
}

export function getMarketOverviewApi(params: MarketOverviewParams) {
  return requestClient.get('/v1/market/overview', { params });
}

export function getMarketTrendApi(params: MarketTrendParams) {
  return requestClient.get('/v1/market/trend', { params });
}

export function getMarketYearlyApi(params: MarketYearlyParams) {
  return requestClient.get('/v1/market/yearly', { params });
}

File: backend/backend/schemas/market.py
from pydantic import BaseModel, Field


class OverviewQuery(BaseModel):
    year: int = Field(..., ge=2000, le=2100)
    month: int = Field(..., ge=0, le=12)
    data_type: str = Field("retail", pattern="^(retail|production)$")
    date_type: str = Field("monthly", pattern="^(monthly|quarterly|yearly)$")
    level_type: str = Field("all", pattern="^(all|nev|bev)$")


class TrendQuery(BaseModel):
    years: int = Field(3, ge=1, le=10)
    granularity: str = Field("monthly", pattern="^(monthly|yearly)$")
    data_type: str = Field("retail", pattern="^(retail|production)$")
    date_type: str = Field("monthly", pattern="^(monthly|quarterly|yearly)$")
    level_type: str = Field("all", pattern="^(all|nev|bev)$")


class YearlyQuery(BaseModel):
    year: int = Field(..., ge=2000, le=2100)
    data_type: str = Field("retail", pattern="^(retail|production)$")
    date_type: str = Field("monthly", pattern="^(monthly|quarterly|yearly)$")
    level_type: str = Field("all", pattern="^(all|nev|bev)$")